src/diffusers/models/adapter.py

```diff
@@ -0,0 +1,244 @@
+from typing import List, Optional
+
+import torch
+import torch.nn as nn
+
+from ..configuration_utils import ConfigMixin, register_to_config
+from .modeling_utils import ModelMixin
+from .resnet import Downsample2D
+
+
+class BottleneckResnetBlock(nn.Module):
+    def __init__(self, in_c, mid_c, out_c, down, ksize=3, sk=False, use_conv=True, proj_ksize=1):
+        super().__init__()
+        ps = ksize // 2
+        proj_pad = proj_ksize // 2
+
+        if in_c != mid_c or sk is False:
+            self.in_conv = nn.Conv2d(in_c, mid_c, proj_ksize, 1, proj_pad)
+        else:
+            self.in_conv = None
+
+        if out_c != mid_c:
+            self.out_conv = nn.Conv2d(mid_c, out_c, proj_ksize, 1, proj_pad)
+        else:
+            self.out_conv = None
+
+        self.block1 = nn.Conv2d(mid_c, mid_c, 3, 1, 1)
```

Comment:
this line and the two following lines is different to what we have currently in the UNet and why it can be called "BottleNeck". Nevertheless the rest is exactly the same as far as I can see